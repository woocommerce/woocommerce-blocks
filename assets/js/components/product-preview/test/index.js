/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import ProductPreview from '../';

jest.mock( '@woocommerce/block-settings', () => ( {
	PLACEHOLDER_IMG_SRC: 'placeholder.png',
	THUMBNAIL_SIZE: 300,
} ) );

describe( 'ProductPreview', () => {
	test( 'should render a single product preview with an image', () => {
		const product = {
			id: 1,
			name: 'Winter Jacket',
			price_html:
				'<span class="woocommerce-Price-amount amount"><span class="woocommerce-Price-currencySymbol">&#36;</span>65.00</span>',
			images: [
				{
					src: 'https://example.local/product.jpg',
				},
			],
		};
		const component = TestRenderer.create( <ProductPreview product={ product } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render a single product preview without an image', () => {
		const product = {
			id: 1,
			name: 'Winter Jacket',
			price_html:
				'<span class="woocommerce-Price-amount amount"><span class="woocommerce-Price-currencySymbol">&#36;</span>65.00</span>',
			images: [],
		};
		const component = TestRenderer.create( <ProductPreview product={ product } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
