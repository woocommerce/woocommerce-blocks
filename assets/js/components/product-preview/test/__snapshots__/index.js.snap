// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPreview should render a single product preview with an image 1`] = `
<div
  className="wc-product-preview wc-block-grid__product"
>
  <div
    className="wc-product-preview__image wc-block-grid__product-image"
  >
    <img
      alt=""
      className="wc-product-preview__image"
      src="https://example.local/product.jpg"
      style={
        Object {
          "width": "300px",
        }
      }
    />
  </div>
  <div
    className="wc-product-preview__title wc-block-grid__product-title"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Winter Jacket",
      }
    }
  />
  <div
    className="wc-product-preview__price wc-block-grid__product-price"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<span class=\\"woocommerce-Price-amount amount\\"><span class=\\"woocommerce-Price-currencySymbol\\">&#36;</span>65.00</span>",
      }
    }
  />
  <span
    className="wp-block-button"
  >
    <span
      className="wc-product-preview__add-to-cart wc-block-grid__product-add-to-cart wp-block-button__link"
    >
      Add to cart
    </span>
  </span>
</div>
`;

exports[`ProductPreview should render a single product preview without an image 1`] = `
<div
  className="wc-product-preview wc-block-grid__product"
>
  <div
    className="wc-product-preview__image wc-block-grid__product-image"
  >
    <img
      alt=""
      className="wc-product-preview__image"
      src="placeholder.png"
      style={
        Object {
          "width": "300px",
        }
      }
    />
  </div>
  <div
    className="wc-product-preview__title wc-block-grid__product-title"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Winter Jacket",
      }
    }
  />
  <div
    className="wc-product-preview__price wc-block-grid__product-price"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<span class=\\"woocommerce-Price-amount amount\\"><span class=\\"woocommerce-Price-currencySymbol\\">&#36;</span>65.00</span>",
      }
    }
  />
  <span
    className="wp-block-button"
  >
    <span
      className="wc-product-preview__add-to-cart wc-block-grid__product-add-to-cart wp-block-button__link"
    >
      Add to cart
    </span>
  </span>
</div>
`;
